src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -549,90 +553,75 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 
 		// If group still isn't defined because of a disabled override we resolve it
 		if (!group) {
-			const [resolvedGroup, activation] = this.findTargetGroup(typedEditor, options, isOpenInEditorGroup(optionsOrPreferredGroup) ? optionsOrPreferredGroup : preferredGroup);
-			if (activation) {
-				options = { ...options, activation };
-			}
+			([group, activation] = this.findTargetGroup({ editor: typedEditor, options }, preferredGroup));
+		}
 
-			group = resolvedGroup;
+		// Mixin editor group activation if any
+		if (activation) {
+			options = { ...options, activation };
 		}
 
 		return group.openEditor(typedEditor, options);
 	}
 
-	private async doResolveEditor(editor: IEditorInputWithOptions | IUntypedEditorInput, preferredGroup?: OpenInEditorGroup): Promise<[ReturnedOverride, IEditorGroup | undefined, IEditorOptions | undefined]> {
+	private async doResolveEditor(editor: IEditorInputWithOptions | IUntypedEditorInput, preferredGroup: OpenInEditorGroup | undefined): Promise<[ReturnedOverride, IEditorGroup | undefined, EditorActivation | undefined]> {
```

Comment:
This method got simpler by returning the `EditorActivation` and not `IEditorOptions` to reduce ambiguity. Because if this method returns `IUntypedEditor` which has options itself and options as another param, which options object am I supposed to use?

Reply:
Maybe I don't understand, but this function returns an `IEditorInput`, not an `IUntypedEditor` and therefore there are no options here

Reply:
Correction, it returns an `IEditorInputWithOptions` so maybe it makes more sense to drop the options / activation pass back all together?

Reply:
I don't understand. It may return `ReturnedOverride.None` and `ReturnedOverride.Abort`, so I think we cannot drop the explicit return of `EditorActivation`?

Reply:
Ah, for `ABORT` it doesn't matter because we just skip that editor, but I guess for the `NONE` case we still used the returned group + activation for mixing in.